templates/case/_reschedule_finish_hearing_deadline.html.twig

```diff
@@ -1,23 +1,29 @@
 {% trans_default_domain 'case' %}
 
-<div class="modal fade" id="reschedule_finish_hearing_deadline{{ case.id }}">
+<div class="modal fade" id="reschedule_finish_hearing_deadline">
     <div class="modal-dialog">
         <div class="modal-content">
+            {{ form_start(reschedule_form, {'method': 'POST', 'action': path('case_reschedule_finish_hearing_deadline', {'id': case.id}), 'attr': {'id': 'reschedule_hearing_form'}}) }}
             <div class="modal-header">
                 <h5 class="modal-title" id="rescheduleFinishHearingDeadlineModalLabel">{% trans %}Reschedule finish hearing deadline{% endtrans %}</h5>
                 <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                     <span aria-hidden="true"><i class="fas fa-times"></i></span>
                 </button>
             </div>
             <div class="modal-body">
-                {{ form_start(reschedule_form, {'action': path('case_reschedule_finish_hearing_deadline', {'id': case.id})}) }}
                 {{ form_widget(reschedule_form) }}
             </div>
             <div class="modal-footer">
                 <button type="button" class="btn btn-secondary" data-dismiss="modal">{% trans %}Close{% endtrans %}</button>
-                <button type="submit" class="btn btn-primary">{% trans %}Reschedule{% endtrans %}</button>
-                {{ form_end(reschedule_form) }}
+                <button id="finish_hearing_submit_button" type="submit" class="btn btn-primary">{% trans %}Reschedule{% endtrans %}</button>
```

Comment:
It may be a small thing, bit the id `finish_hearing_submit_button` is not really describing this button. Maybe something like `reschedule_hearing_deadline_submit_button` is better?

Reply:
We can actually do without an id on the submit button if we refactor the JavaScript code a bit:

```js
$(() => {
  const $form = $('#reschedule_hearing_form')
  $form.find('[type="submit"]').on('click', …)
})
```

Using an even more consistent naming scheme we can generalize the “submit on modal” code. The names `finish_hearing_submit_button`, `reschedule_hearing_form` and `reschedule_finish_hearing_deadline` used in assets/case/reschedule_finish_hearing_deadline.js don't have much in common. They should probably have a common prefix.

Reply:
Generalized in fe66e0a516dc35b6d759cb463b1b43772aaf1e07.